apps/editing-toolkit/editing-toolkit-plugin/dotcom-fse/lib/site-intent/use-site-intent.js

```diff
@@ -1,18 +1,28 @@
 import apiFetch from '@wordpress/api-fetch';
-import { useState, useEffect, useCallback } from '@wordpress/element';
+import { useQuery } from 'react-query';
 
 const useSiteIntent = () => {
-	const [ siteIntent, setSiteIntent ] = useState( '' );
-
-	const fetchSiteIntent = useCallback( () => {
-		apiFetch( { path: '/wpcom/v2/site-intent' } )
-			.then( ( result ) => setSiteIntent( result.site_intent ) )
-			.catch( () => setSiteIntent( '' ) );
-	}, [] );
-
-	useEffect( () => {
-		fetchSiteIntent();
-	}, [ fetchSiteIntent ] );
-	return siteIntent;
+	const queryInfo = useQuery(
+		'site-intent', // add Site ID ?
+		() =>
+			apiFetch( { path: '/wpcom/v2/site-intent' } )
+				.then( ( result ) => result.site_intent )
+				.catch( () => '' ),
+		{
+			refetchOnWindowFocus: false,
+			staleTime: Infinity,
+			//enabled: !! siteId,
+		}
+	);
+	return queryInfo.data || '';
 };
 export default useSiteIntent;
+
+/*
+	different endpoint / request mechanism used outside of editing-toolkit:
```

Comment:
I started thinking that perhaps we could move this to `@automattic./data-stores` package and then I noticed we already have intent available at the onboard data store — using react-query no-less:

https://github.com/Automattic/wp-calypso/blob/2c57c5129969afbbcb7034e2edf8f143bf2589c9/packages/data-stores/src/queries/use-site-intent.ts#L4-L20

https://github.com/Automattic/wp-calypso/blob/2c57c5129969afbbcb7034e2edf8f143bf2589c9/packages/data-stores/src/onboard/index.ts#L42

It's a bit more complicated as it adds `@wordpress/data` layer where the caching happens, but it's less code overall since it's shared, and since it's WP-data it's wp-admin compatible.